src/jit/lsra.cpp

```diff
@@ -5686,12 +5684,37 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
             // the next ref, remember it.
             else if ((bestScore & UNASSIGNED) != 0 && intervalToUnassign != nullptr)
             {
+#ifdef _TARGET_ARM_
+                // We should update a double register if physRegToUnassign is part of the double register
+                if (intervalToUnassign->registerType == TYP_DOUBLE &&
+                    !genIsValidDoubleReg(physRegToUnassign->regNum))
+                    physRegToUnassign = findAnotherHalfRegRec(physRegToUnassign);
+#endif
                 updatePreviousInterval(physRegToUnassign, intervalToUnassign, intervalToUnassign->registerType);
+#ifdef _TARGET_ARM_
```

Comment:
@CarolEidt 
I did like what you said.
```
#ifdef _TARGET_ARM_
        if ((bestScore & UNASSIGNED) == 0)
#else
        if (intervalToUnassign != nullptr)
#endif
```
But in below code that checks if it is UNASSIGNED or not, 
``` if (assignedInterval == nullptr || assignedInterval->getNextRefLocation() > lastLocation) ```
and
``` if (anotherRegRec->assignedInterval == nullptr || anotherRegRec->assignedInterval->getNextRefLocation() > lastLocation)```
are not guaranteed that assignedInterval is nullptr.
```
bool LinearScan::isUnassigned(RegRecord* regRec, LsraLocation lastLocation, Interval* currentInterval)
{
    Interval* assignedInterval = regRec->assignedInterval;

    if (assignedInterval == nullptr ||
        assignedInterval->getNextRefLocation() > lastLocation)
    {
#ifdef _TARGET_ARM_
        if (currentInterval != nullptr && currentInterval->registerType == TYP_DOUBLE)
        {
            RegRecord* anotherRegRec = findAnotherHalfRegRec(regRec);

            if (anotherRegRec->assignedInterval == nullptr ||
                anotherRegRec->assignedInterval->getNextRefLocation() > lastLocation)
            {
                // In case the currentInterval has a double register,
                // the score would be set UNASSIGNED if another register is also not set.
                return true;
            }
        }
        else
#endif
            return true;
    }
    return false;
}
```
So, There are cases to throw into the else case without being unassigned.